hmvec/hmvec.py

```diff
@@ -255,8 +273,76 @@ def add_battaglia_profile(self,name,family=None,param_override={},
+            elif key in battaglia_defaults[family]:
+                pparams[key] = param_override[key]
+            else:
+                raise ValueError # param in param_override doesn't seem to be a Battaglia parameter
+
+        # Convert masses to m200critz
+        rhocritz = self.rho_critical_z(self.zs)
+        if self.mdef=='vir':
+            delta_rhos1 = rhocritz*self.deltav(self.zs)
+        elif self.mdef=='mean':
+            delta_rhos1 = self.rho_matter_z(self.zs)*200.
+        rvirs = self.rvir(self.ms[None,:],self.zs[:,None])
+        cs = self.concentration()
+        delta_rhos2 = 200.*self.rho_critical_z(self.zs)
+        m200critz = mdelta_from_mdelta(self.ms,cs,delta_rhos1,delta_rhos2)
+        r200critz = R_from_M(m200critz,self.rho_critical_z(self.zs)[:,None],delta=200.)
+
+        # Generate profiles
+        """
+        The physical profile is rho(r) = f(2r/R200)
+        We rescale this to f(x), so x = r/(R200/2) = r/rgs
+        So rgs = R200/2 is the equivalent of rss in the NFW profile
+        """
+        omb = self.p['ombh2'] / self.h**2.
+        omm = self.om0
+        presFunc = lambda x: P_e_generic_x(x,m200critz[...,None],r200critz[...,None],self.zs[:,None,None],omb,omm,rhocritz[...,None,None],
+                                    alpha=pparams['battaglia_pres_alpha'],
+                                    gamma=pparams['battaglia_pres_gamma'],
+                                    P0_A0=pparams['P0_A0'],
+                                    P0_alpham=pparams['P0_alpham'],
+                                    P0_alphaz=pparams['P0_alphaz'],
+                                    xc_A0=pparams['xc_A0'],
+                                    xc_alpham=pparams['xc_alpham'],
+                                    xc_alphaz=pparams['xc_alphaz'],
+                                    beta_A0=pparams['beta_A0'],
+                                    beta_alpham=pparams['beta_alpham'],
+                                    beta_alphaz=pparams['beta_alphaz'])
+
+        # This function returns the pressure in units of 1/Mpc/s^2
+
+        rgs = r200critz
+        cgs = rvirs/rgs
+
+        sigmaT=constants.physical_constants['Thomson cross section'][0] # units m^2
+        mElect=constants.physical_constants['electron mass'][0] / default_params['mSun']# units kg
+        # c^2 and sigma cancel units so dont rescale.
+        # There is a G in pressure, units m3/kg/s2. But converted to Mpc, mSun, s
+        # P_200 = G M rho_crit 200 /(2* r_200)
+
+        ks,pkouts = generic_profile_fft(presFunc,cgs,rgs[...,None],self.zs,self.ks,xmax,nxs,doMassNorm=False)
+        self.pk_profiles[name] = pkouts.copy()*4*np.pi*(sigmaT/(mElect*constants.c**2))*(r200critz**3*((1+self.zs)**2/self.h_of_z(self.zs))[...,None])[...,None]# *(r200critz**3*((1+self.zs)**2/self.comoving_radial_distance(self.zs)**1)[...,None])[...,None] # *(r200critz**3*((1+self.zs)**2/self.comoving_radial_distance(self.zs)**3)[...,None])[...,None]
```

Comment:
Has the unit mismatch been fixed? If yes, please remove the comment.

Reply:
Yes. The comments have been removed